Makefile

```diff
@@ -84,7 +86,7 @@ endif
 
 $(MACOS_PROJ_PATH): $(BUILD_DEPS) $(MACOS_USER_DATA_PATH)/WorkspaceSettings.xcsettings
 	mkdir -p $(MACOS_OUTPUT_PATH)
-	(cd $(MACOS_OUTPUT_PATH) && cmake -G Xcode ../.. \
+	(cd $(MACOS_OUTPUT_PATH) && $(CMAKE) -G Xcode ../.. \
```

Comment:
This change makes sense, but CMake already defines a variable for that:

https://cmake.org/cmake/help/latest/variable/CMAKE_COMMAND.html

Reply:
It's correct that CMake defines a variable to invoke the cmake command from within a running CMake build configuration. This variable is unknown to the Make system that invokes CMake in this case.

What this patch is aiming for is the ability to bootstrap a Mapbox build like:
`CMAKE=$BREW_HOME/bin/cmake && make macos`